src/lib/core/config.js

```diff
@@ -34,37 +33,75 @@ var Config = function(options) {
   this.shownNoAccountConfigMsg = false; // flag to ensure "no account config" message is only displayed once to the user
   this.corsParts = [];
   this.providerUrl = null;
-  this.events.setCommandHandler("config:cors:add", (url) => {
+
+  this.registerEvents();
+};
+
+Config.prototype.registerEvents = function() {
+  this.events.setCommandHandler("config:cors:add", (url, cb = () => {}) => {
     this.corsParts.push(url);
     this._updateBlockchainCors();
-  });
 
+    let configFilePath = this._getFileOrOject(this.configDir, 'blockchain', 'blockchain');
+    const completeConfig = this._mergeConfig(configFilePath, {}, null, true);
+    const envConfig = completeConfig[this.env];
+
+    if (!envConfig.rpcCorsDomain || envConfig.rpcCorsDomain === 'auto') {
+      envConfig.rpcCorsDomain = {auto: true, additionalCors: []};
+    } else if (typeof envConfig.rpcCorsDomain === 'string') {
+      envConfig.rpcCorsDomain = {auto: false, additionalCors: envConfig.rpcCorsDomain.join(',')};
+    } else if (!envConfig.rpcCorsDomain.additionalCors) {
+      envConfig.rpcCorsDomain.additionalCors = [];
+    }
+    if (!envConfig.wsOrigins || envConfig.wsOrigins === 'auto') {
+      envConfig.wsOrigins = {auto: true, additionalCors: []};
+    } else if (typeof envConfig.wsOrigins === 'string') {
+      envConfig.wsOrigins = {auto: false, additionalCors: envConfig.wsOrigins.join(',')};
+    } else if (!envConfig.wsOrigins.additionalCors) {
+      envConfig.wsOrigins.additionalCors = [];
+    }
+
+    envConfig.rpcCorsDomain.additionalCors.push(url);
+    envConfig.wsOrigins.additionalCors.push(url);
+
+    let configString = JSON.stringify(completeConfig, null, 2);
+    if (fs.existsSync(configFilePath + '.js')) {
+      configFilePath += '.js';
+      // TODO Find a better way, because if the users have more than the config, this doesn't work
+      configString = 'module.exports = ' + configString;
+    } else {
+      configFilePath += '.json';
+    }
 
-  self.events.setCommandHandler("config:contractsConfig", (cb) => {
-    cb(self.contractsConfig);
+    fs.writeFile(configFilePath, configString, cb);
+    this.logger.info(__('Added "%s" to the CORS. You will need to restart your blockchain node or Embark for the changes to take effect', url));
   });
```

Comment:
Here I think we should also call `this._updateBlockchainCors();` as the updated CORS values in the config were being used by the `embark-status` plugin.
```suggestion
    this._updateBlockchainCors();
  });
```